pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -295,13 +317,11 @@ func (a Adapter) pushLocal(path string, files []string, delFiles []string, isFor
 	}
 
 	podSelector := fmt.Sprintf("component=%s", a.ComponentName)
-	watchOptions := metav1.ListOptions{
-		LabelSelector: podSelector,
-	}
+
 	// Wait for Pod to be in running state otherwise we can't sync data to it.
-	pod, err := a.Client.WaitAndGetPod(watchOptions, corev1.PodRunning, "Waiting for component to start")
+	pod, err := a.waitAndGetComponentPod(false)
```

Comment:
Can we just use the pod retrieved above after createOrUpdateComponent and rollout check https://github.com/openshift/odo/pull/2811/files#diff-825a1a88d75e789899c58e89a79eb767R73? 

Because that is essentially the new pod, we dont necessarily need to retrieve it again here. You can probably skip the if condition `if componentExists` and initialize `podName := ""` and you wont need this retrieval here.

Reply:
The object returned from `WaitForDeploymentRollout` is a Deployment which we can't get the pod name from directly, unfortunately.